app/vmui/packages/vmui/src/pages/ExploreAlerts/ExploreRules.tsx

```diff
@@ -0,0 +1,244 @@
+import { FC, useEffect, useState, useCallback } from "preact/compat";
+import { useNavigate, useLocation, useSearchParams } from "react-router";
+import { useRulesSetQueryParams as useSetQueryParams } from "./hooks/useSetQueryParams";
+import Spinner from "../../components/Main/Spinner/Spinner";
+import Alert from "../../components/Main/Alert/Alert";
+import Accordion from "../../components/Main/Accordion/Accordion";
+import { useFetchGroups } from "./hooks/useFetchGroups";
+import "./style.scss";
+import RulesHeader from "../../components/ExploreAlerts/RulesHeader";
+import GroupHeader from "../../components/ExploreAlerts/GroupHeader";
+import Rule from "../../components/ExploreAlerts/Rule";
+import ExploreRule from "../../pages/ExploreAlerts/ExploreRule";
+import ExploreAlert from "../../pages/ExploreAlerts/ExploreAlert";
+import ExploreGroup from "../../pages/ExploreAlerts/ExploreGroup";
+import { Group, Rule as APIRule } from "../../types";
+import { getQueryStringValue } from "../../utils/query-string";
+import { getState, getStates, getChanges } from "./helpers";
+import debounce from "lodash.debounce";
+
+const defaultTypesStr = getQueryStringValue("types", "") as string;
+const defaultTypes = defaultTypesStr.split("&").filter((rt) => rt) as string[];
+const defaultStatesStr = getQueryStringValue("states", "") as string;
+const defaultStates = defaultStatesStr.split("&").filter((s) => s) as string[];
+const defaultSearchInput = getQueryStringValue("search", "") as string;
+
+const ExploreRules: FC = () => {
+  const groupId = getQueryStringValue("group_id", "") as string;
+  const ruleId = getQueryStringValue("rule_id", "") as string;
+  const alertId = getQueryStringValue("alert_id", "") as string;
+
+  const [searchInput, setSearchInput] = useState(defaultSearchInput);
+  const [types, setTypes] = useState(defaultTypes);
+  const [states, setStates] = useState(defaultStates);
+  const [modalOpen, setModalOpen] = useState(false);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  if (!location.hash && groupId) {
+    setModalOpen(true);
+  } else {
+    setModalOpen(false);
+  }
+
```

Comment:
move it to `useEffect` 
```suggestion
  useEffect(() => {
    if (!location.hash && groupId) {
      setModalOpen(true);
    } else {
      setModalOpen(false);
    }
  }, [location.hash, groupId]);

```